src/python/pants/python/python_setup.py

```diff
@@ -204,16 +225,28 @@ def artifact_cache_dir(self):
     def scratch_dir(self):
         return os.path.join(self.get_options().pants_workdir, *self.options_scope.split("."))
 
-    def compatibility_or_constraints(self, compatibility):
-        """Return either the given compatibility, or the interpreter constraints. If interpreter
-        constraints are supplied by the CLI flag, return those only.
+    def compatibility_or_constraints(
```

Comment:
We've got a bit of a terminology kludge right now: we have global "interpreter constraints" that can be overridden by per-target "compatibility", and now global "requirement constraints" that can be overridden by per-target "constraints". And "constraints" in the pip world means "requirement constraints". 

Probably we should deprecate "interpreter constraints" and rename to "interpreter compatibility" in some future change?